compiler/src/dotty/tools/dotc/ast/Trees.scala

```diff
@@ -112,9 +112,9 @@ object Trees {
      *  type. (Overridden by empty trees)
      */
     def withType(tpe: Type)(implicit ctx: Context): ThisTree[Type] = {
-      if (tpe.isInstanceOf[ErrorType])
-        assert(ctx.mode.is(Mode.Interactive) || ctx.reporter.errorsReported)
-      else if (Config.checkTreesConsistent)
+      if (Config.checkUnreportedErrors)
+        if (tpe.isInstanceOf[ErrorType]) assert(ctx.reporter.errorsReported)
```

Comment:
Why drop the `ctx.mode.is(Mode.Interactive)`?

Reply:
No need I think. Since Config.checkUnreportedErrors is by default `false`, we don't need the additional test. If someone turns Config.checkUnreportedErrors on for the IDE I assume they know what they are doing.